src/-private/tracking.ts

```diff
@@ -1,48 +1,50 @@
-import { tracked } from '@glimmer/tracking';
-
-class Tag {
-  @tracked private __tagValue__: undefined;
-
-  static consumeTag(tag: Tag): void {
-    // read the tag value
-    tag.__tagValue__;
-  }
-
-  static dirtyTag(tag: Tag): void {
-    // write the tag value
-    tag.__tagValue__ = undefined;
-  }
+import {
+  createStorage,
+  getValue,
+  setValue,
+  TrackedStorage,
+} from 'ember-tracked-storage-polyfill';
+
+export type Tag = TrackedStorage<unknown>;
+
+export interface Node<
+  T extends Array<unknown> | Record<string, unknown> =
+    | Array<unknown>
+    | Record<string, unknown>
+> {
+  collectionTag: Tag | null;
+  tag: Tag | null;
+  tags: Record<string, Tag>;
+  children: Record<string, Node>;
+  proxy: T;
+  value: T;
 }
 
+const neverEq = () => false;
+
 export function createTag(): Tag {
-  return new Tag();
+  return createStorage(null, neverEq);
 }
-
-export const consumeTag = Tag.consumeTag;
-export const dirtyTag = Tag.dirtyTag;
-
-////////////
-
-interface Node {
-  collectionTag: Tag | undefined;
-  proxy: object;
+export const consumeTag = getValue;
+export function dirtyTag(tag: Tag): void {
+  setValue(tag, null);
 }
 
 export let consumeCollection = (node: Node): void => {
   let tag = node.collectionTag;
 
-  if (tag === undefined) {
-    tag = node.collectionTag = createTag();
+  if (tag === null) {
+    tag = node.collectionTag = createStorage(null, neverEq);
   }
 
-  consumeTag(tag);
+  getValue(tag);
 };
 
 export let dirtyCollection = (node: Node): void => {
   const tag = node.collectionTag;
 
-  if (tag !== undefined) {
-    dirtyTag(tag);
+  if (tag !== null) {
+    setValue(tag, null);
   }
 };
 
```

Comment:
I tried to get rid off the following lines (*could not place the comment right there as Github does not allow this*), as this relies on the Ember global, see #113. If this was a regular Ember addon (see other comment), we could import Ember, which should fix this issue.

On the other hand, I wonder why this is actually needed? Without it, 3 tests are failing. @pzuraq can you explain the magic behind dirtying `'[]'` here? And why is this not needed in other places, like https://github.com/tracked-tools/tracked-built-ins/blob/master/addon/-private/array.ts?